src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -62,13 +63,32 @@
     private final String apiUri;
     private String pattern = ".*";
 
+    private @Nonnull String includes = GitHubSCMSource.DescriptorImpl.defaultIncludes;
+    private @Nonnull String excludes = GitHubSCMSource.DescriptorImpl.defaultExcludes;
```

Comment:
You forgot to use a `readResolve` to set these for existing organization folders upon plugin upgrade.

Anyway the approach you have here will cause the default values to be written into every `config.xml`.

Suggestion from [here](https://github.com/jenkinsci/pipeline-plugin/blob/72a4ac9c280013c8a68a7b8b24b9b1a8d6ea045a/DEVGUIDE.md#handling-default-values): leave the _fields_ `@CheckForNull`, null by default, and have the getters read

```java
return includes != null ? includes : GitHubSCMSource.DescriptorImpl.defaultIncludes;
```

and the setters

```java
this.includes = includes.equals(GitHubSCMSource.DescriptorImpl.defaultIncludes) ? null : includes;
```

Then `config.xml` stays clean unless and until you change the defaults, and old folder configurations are loaded smoothly.